sklearn/ensemble/gradient_boosting.py

```diff
@@ -965,8 +964,41 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # fit initial model - FIXME make sample_weight optional
             self.init_.fit(X, y, sample_weight)
 
-            # init predictions
-            y_pred = self.init_.predict(X)
+            if is_classifier(self.init_): 
+                n_classes = np.unique(y).shape[0]
```

Comment:
Maybe the local variable `n_classes` should be renamed `n_targets` to make is explicit that this can also work for regression.